src/sentry/interfaces/stacktrace.py

```diff
@@ -837,48 +931,61 @@ def to_json(self):
             'registers': self.registers,
         })
 
-    def get_hash(self, platform=None, variant='system'):
-        if variant not in ('app', 'system'):
-            return []
-
+    def get_grouping_component(self, platform=None, variant=None):
         frames = self.frames
+        contributes = None
+        hint = None
+        all_frames_are_in_app = False
 
         # TODO(dcramer): this should apply only to platform=javascript
         # Browser JS will often throw errors (from inlined code in an HTML page)
         # which contain only a single frame, no function name, and have the HTML
         # document as the filename. In this case the hash is often not usable as
         # the context cannot be trusted and the URL is dynamic (this also means
         # the line number cannot be trusted).
-        stack_invalid = (len(frames) == 1 and not frames[0].function and frames[0].is_url())
-
-        if stack_invalid:
-            return []
-
-        system_frames = variant == 'system'
-        if not system_frames:
+        if (len(frames) == 1 and not frames[0].function and frames[0].is_url()):
+            contributes = False
+            hint = 'ignored single frame stack'
+        elif variant == 'app':
             total_frames = len(frames)
-            frames = [f for f in frames if f.in_app] or frames
+            in_app_count = sum(1 if f.in_app else 0 for f in frames)
+            if in_app_count == 0:
+                in_app_count = total_frames
+                all_frames_are_in_app = True
```

Comment:
The variable name `all_frames_are_in_app` is confusing and misleading when reading expressions like `not frame.in_app and not all_frames_are_in_app` below. Maybe `treat_ all_frames_as_in_app` or `condsider_all_frames_in_app` would be better?

Reply:
Indeed.